server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -669,21 +669,31 @@ private void stopBeingLeader()
 
   private List<CoordinatorDuty> makeHistoricalManagementDuties()
   {
-    return ImmutableList.of(
-        new LogUsedSegments(),
+    List<CoordinatorDuty> duties = new ArrayList<>();
+    if (config.isLogUsedSegmentsDutyEnabled()) {
+      duties.add(new LogUsedSegments());
+    }
+    duties.addAll(ImmutableList.of(
         new UpdateCoordinatorStateAndPrepareCluster(),
         new RunRules(DruidCoordinator.this),
         new UnloadUnusedSegments(),
         new MarkAsUnusedOvershadowedSegments(DruidCoordinator.this),
         new BalanceSegments(DruidCoordinator.this),
         new EmitClusterStatsAndMetrics(DruidCoordinator.this)
+    ));
+    log.debug(
+        "Done making historical management duties %s",
+        duties.stream().map(duty -> duty.getClass().getName()).collect(Collectors.toList())
     );
+    return duties;
```

Comment:
nit: `Collections.unmodifiableList(duties)`

Reply:
makeIndexingServiceDuties below used ImmutableList.copyOf(duties) so I will stick with that for consistency

Reply:
reverting to the original way of doing this now that LogUsedSegments is always used again